File: LeetCode/python/customSortString.py
from collections import Counter

def customSortString(order, s): 
  counter = Counter(s)
  res = ''
  
  for c in order:
    if c in counter:
      res += c * counter[c]
      del counter[c]
      
  for c in counter:
    res += c * counter[c]
    
  print(res)

  return res


order = "bcafg"
s = "abcd" 

customSortString(order, s)

File: LeetCode/java/CustomSortString.java
import java.util.Arrays;

public class CustomSortString {
    public static void main(String[] args) {

    }

    public String customSortString(String order, String s) {
        int lenS = s.length();
        Character[] res = new Character[lenS];
        for (int i = 0; i < lenS; i++) {
            res[i] = s.charAt(i);
        }

        // Define the custom comparator
        Arrays.sort(res, (c1, c2) -> {
            // The index of the character in order determines the value to be sorted by
            return order.indexOf(c1) - order.indexOf(c2);
        });

        // Return the res
        String resString = "";
        for (Character c: res) {
            resString += c;
        }
        return resString;
    }
}
